EIPS/eip-network_upgrade_windows.md

```diff
@@ -0,0 +1,171 @@
+
+### Roadmap Network Upgrades
+
+Roadmap network upgrades are network upgrades that are deliberate and measured
+to improve the protocol and ecosystem. Historical examples are Homestead,
+Byzantium, and Constantinople.
+
+Roadmap network upgrades should be scheduled in one of four windows: the third
+week of January, the third week fo April, the third week of July or the third
+week of October. When initiating a network upgrade or early in the planning
+process a particular window should be targeted. A block number for the network
+upgrade should be chosen four to six weeks prior to the network upgrade window.
+Scheduling details such as whether this choice is made prior to or after testnet
+deployment are out of scope of this EIP.
+
+> **Note to reviewers:** The months and week chosen are to provide an initial
+> recommendation and are easily modifiable prior to final call. They thread the
+> needle between many third quarter and fourth quarter holidays.
+
+A roadmap upgrade typically should not occur in adjacent upgrade windows. Hence
+if a roadmap upgrade occurred in April then the July window would not be used
+for network upgrades.
+
+If a planned roadmap upgrade needs to be rescheduled then strong consideration
+should be given to rescheduling the upgrade for the next window in three months
+time.
+
+Not every upgrade window must be used.
+
+### Priority Network Upgrades
+
+Priority network upgrades are reserved for upgrades that require more urgency
+than a roadmap network upgrade yet do not present a systemic risk to the network
+or the ecosystem. To date there have been no examples of a priority upgrade.
+Possible examples may include roadmap upgrades that need to occur in multiple
+upgrades or for security risks that have a existing mitigation in place that
+would be better served by a network upgrade. Another possible reason may be to
+defuse the difficulty bomb due to postponed roadmap upgrades.
+
+Priority network upgrades are best launched in unused roadmap launch windows,
+namely the third week of January, April, July, and October. If necessary they
+may be launched in the third week of any month, but strong consideration and
+preference should be given to unused roadmap launch windows.
+
+Priority network upgrades should be announced and a block chosen far enough in
+advance so major clients implementors can release software with the needed block
+number in a timely fashion. It is expected that two to four weeks would be
+sufficient.
+
+### Critical Network Upgrades
+
+Critical network upgrades are network upgrades that are designed to address
+systemic risks to the protocol or to the ecosystem. Historical examples include
+Dao Fork, Tangerine Whistle, and Spurious Dragon.
+
+This EIP provides neither guidance nor restrictions to the development and
+deployment of these emergency hard forks. These upgrades are typically launched
+promptly after a solution to the systemic risk is agreed upon between the client
+implementors.
```

Comment:
I would suggest guidance that such upgrades perform the minimal amount of work possible to mitigate the issue, but that is also probably outside the scope here.

Reply:
done